socket/inet_pton.c

```diff
@@ -0,0 +1,240 @@
+/* Copyright (C) 1996-2018 Free Software Foundation, Inc.
```

Comment:
so we usually split code ports like this.
1. Original code with commit author as the original author, maintainer of the project or foundation.
2. Your modifications on a *separate* commit on top of the original commit.